azerothcore/azerothcore-wotlk
# feat(Core/ChatHandler): Flood Control Addon Messages

* Flood control addon messages (anti-spam / rate limit)

* cherry-pick commit (https://github.com/TrinityCore/TrinityCore/commit/5384fc194f9061bbac314abb66a9bac88aaaf760)

* cherry-pick commit (https://github.com/stoneharry/TrinityCore/commit/a0629fc704878378afcbbf00e9bd83b44a40f8f8)

Co-Authored-By: stoneharry <3818405+stoneharry@users.noreply.github.com>
Co-Authored-By: Shauren <shauren.trinity@gmail.com>



## Changes Proposed:
-  flood control for addon messages
-  Allow to track addon messages in metric

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- none
- 


## How to Test the Changes:


1. Run `/script for i=0,1000 do SendClientRequest("feaihf", 5, 1); end` for ~5s to give a build up on addon messages
2. see how long it takes to process all the packets
3. For in detail about it read this thread https://github.com/TrinityCore/TrinityCore/pull/26908


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
